extensions/wikia/AdEngine/js/template/bfab.js

```diff
@@ -1,18 +1,53 @@
 /*global define*/
 define('ext.wikia.adEngine.template.bfab', [
+	'ext.wikia.adEngine.domElementTweaker',
 	'ext.wikia.adEngine.slotTweaker',
+	'wikia.browserDetect',
 	'wikia.log',
 	'wikia.document'
-], function (slotTweaker, log, doc) {
+], function (DOMElementTweaker, slotTweaker, browser, log, doc) {
 	'use strict';
 
 	var logGroup = 'ext.wikia.adEngine.template.bfab',
 		slot = doc.getElementById('BOTTOM_LEADERBOARD') || doc.getElementById('MOBILE_BOTTOM_LEADERBOARD');
 
+
+	function isRecoveryNotSupportedBrowser() {
+		return browser.isIE() || browser.isEdge();
+	}
+
+	function mirrorDOMStructure(parentNode) {
+		var originalIframe = document.createElement('iframe');
+		var div = document.createElement('div');
+
+		div.appendChild(originalIframe);
+		parentNode.appendChild(div);
+		originalIframe.style.display = 'none';
+		div.style.display = 'none';
+		return originalIframe;
+	}
+
+	function recoverSlot(iframe, params) {
+		var adContainer = iframe.parentNode.parentNode.parentNode.parentNode,
+			originalIframe = mirrorDOMStructure(document.querySelector('#' + params.slotName + ' > div > div'));
+
+		if (isRecoveryNotSupportedBrowser()) {
+			DOMElementTweaker.hide(adContainer, true);
+			return;
+		}
+
+		slotTweaker.tweakRecoveredSlot(originalIframe, iframe);
+	}
+
 	function show(params) {
+		console.log(params);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2228)